mlflow/R/mlflow/.create-test-env.R

```diff
@@ -12,8 +12,7 @@ keras::install_keras(method = "conda", envname = mlflow:::mlflow_conda_env_name(
 reticulate::conda_install("'h5py<3.0.0'", envname = mlflow:::mlflow_conda_env_name(), pip = TRUE)
 reticulate::conda_install(Sys.getenv("MLFLOW_HOME", "../../../../."), envname = mlflow:::mlflow_conda_env_name(), pip = TRUE)
 reticulate::conda_install("xgboost", envname = mlflow:::mlflow_conda_env_name())
-# Pin h2o to prevent version-mismatch between python and R
-reticulate::conda_install("h2o==3.30.1.3", envname = mlflow:::mlflow_conda_env_name(), pip = TRUE)
+reticulate::conda_install("h2o", envname = mlflow:::mlflow_conda_env_name(), pip = TRUE)
```

Comment:
I'm taking a look at https://github.com/mlflow/mlflow/pull/3512 which pinned `h2o` to remember why we needed to pin `h2o`.

Reply:
If we do not specify the h2o version, can we ensure h2o py and R are the same version ?

Reply:
e.g. h2o release a new R-package, but conda side haven't update to the latest version.

Reply:
> h2o release a new R-package, but conda side haven't update to the latest version.

Yep, this might happen.